overrides/kubejs/server_scripts/src/ingot_heatcast.js

```diff
@@ -0,0 +1,363 @@
+    let itemLocation = Utils.id(item)
+    event.addJson(`${itemLocation.namespace}:tfc/item_sizes/${itemLocation.path}.json`, json)
+  }
+  /**
+   *
+   * @param {string} fluid
+   * @param {number} tier
+   * @param {number} meltTemp
+   * @param {number} specificHeatCap
+   * @param {string} ingots
+   * @param {string} sheets
+   */
+  let addTFCMetalFluid = (fluid, tier, meltTemp, specificHeatCap, ingots, sheets, doubleIngots) => {
+    /** @type {Internal.LinkedHashMap} */
+    let json = JsonIO.toObject({
+      tier: tier,
+      fluid: fluid,
+      melt_temperature: meltTemp,
+      specific_heat_capacity: specificHeatCap,
+      ingots: Ingredient.of(ingots).toJson(),
+      sheets: Ingredient.of(sheets).toJson(),
+      double_ingots: Ingredient.of(doubleIngots).toJson()
+    })
+    let itemLocation = Utils.id(fluid)
+    event.addJson(`${itemLocation.namespace}:tfc/metals/${itemLocation.path}.json`, json)
+  }
+  let addTFCCastingRecipe = (item, fluid, amount) => {
+    /** @type {Internal.LinkedHashMap} */
+    let json = JsonIO.toObject({
+      type: "tfc:casting",
+      mold: {
+        item: "tfc:ceramic/ingot_mold"
+      },
+      fluid: {
+        ingredient: fluid,
+        amount: amount
+      },
+      result: {
+        item: item
+      },
+      break_chance: 0.1
+    })
+    let fluidLocation = Utils.id(fluid)
+    event.addJson(`tfc:recipes/casting/${fluidLocation.path}_to_mold.json`, json)
+    json.mold.item = "tfc:ceramic/fire_ingot_mold"
+    json.break_chance = 0.01
+    event.addJson(`tfc:recipes/casting/${fluidLocation.path}_to_fire_mold.json`, json)
+  }
+  gtceuIngots1.forEach((id) => {
+    let fluid = Utils.id("gtceu:" + id)
+    let ingot = Utils.id("gtceu:" + id + "_ingot")
+    let temp = $FluidHelper.getTemperature(Fluid.of(fluid)) - 273 // Kelvin to Celcius
+    temp = Math.max(230, temp) // set at minimum temp like Tin
+    addTFCHeatCapability(ingot.toString(), 2.857, Math.floor(temp * 0.6), Math.floor(temp * 0.8))
+    addTFCMetalFluid(fluid.toString(), 3, temp, 0.0085, "#forge:ingots/" + id, "#forge:plates/" + id, "#forge:double_ingots/" + id)
+    if (temp > 1540) return // skip if temp is over Red/Blue Steel
+    addTFCHeatingRecipe(ingot.toString(), fluid.toString(), temp, 144)
+    addTFCCastingRecipe(ingot.toString(), fluid.toString(), 144)
+  })
+
+  let gtceuFluidCast = [
+    "tin",
+    "silver",
+    "gold",
+    "copper",
+    "nickel",
+    "bismuth",
+    "rubber",
+  ]
+
+  gtceuFluidCast.forEach((id) => {
+    addTFCMetalFluid("gtceu:" + id, 3, Math.max(230, $FluidHelper.getTemperature(Fluid.of("gtceu:" + id)) - 273), 0.0085, "#forge:ingots/" + id, "#forge:plates/" + id, "#forge:double_ingots/" + id)
+    if (id == "gold" || id == "copper") return
+    addTFCCastingRecipe("gtceu:" + id + "_ingot", "gtceu:" + id, 144)
+  })
+  
+  addTFCCastingRecipe("tfc:metal/ingot/copper", "gtceu:copper", 144)
+  addTFCCastingRecipe("tfc:metal/ingot/gold", "gtceu:gold", 144)
```

Comment:
why are these 2 seperate?

Reply:
> why are these 2 seperate?

skill issue on my part, they were done at separate times and went over my head